core/string/ustring.cpp

```diff
@@ -3371,6 +3371,145 @@ float String::similarity(const String &p_string) const {
 	return (2.0f * inter) / sum;
 }
 
+// Based on: https://github.com/ka-weihe/fastest-levenshtein
+int String::levenshtein_distance(const String &p_string) const {
+	auto a = *this;
+	auto b = p_string;
+	if (a.length() < b.length()) {
+		SWAP(a, b);
+	}
+
+	if (b.is_empty()) {
+		return a.length();
+	}
+
+	HashMap<char32_t, uint> peq(0x10000);
```

Comment:
Sizeless `uint` is not a standard C++ type, it should be either `unsigned`/`unsigned int` or `uintXX_t`.